cartographer/mapping_2d/sparse_pose_graph/optimization_problem.cc

```diff
@@ -112,19 +112,19 @@ void OptimizationProblem::AddSubmap(const int trajectory_id,
   CHECK_GE(trajectory_id, 0);
   submap_data_.resize(
       std::max(submap_data_.size(), static_cast<size_t>(trajectory_id) + 1));
-  submap_data_[trajectory_id].push_back(SubmapData{submap_pose});
   trajectory_data_.resize(
       std::max(trajectory_data_.size(), submap_data_.size()));
+
+  auto& trajectory_data = trajectory_data_.at(trajectory_id);
+  submap_data_[trajectory_id].emplace(
+      trajectory_data.next_submap_index++,
```

Comment:
Readability and consistency: Please move the `++trajectory_data.next_submap_index` after the `emplace`.